### PR TITLE
Direct C# interop replacing default serializing in JsonSerializer

### PR DESCRIPTION
**Pull request by ajdust**
See discussion https://github.com/sebastienros/jint/discussions/1228

### CODE REVIEW

File: Jint/Native/Json/JsonSerializer.cs [hunk 1]
```diff
@@ -201,9 +201,12 @@ private JsValue SerializeJSONProperty(JsValue key, JsValue holder)
 
             if (value is ObjectInstance { IsCallable: false } objectInstance)
             {
-                return SerializesAsArray(objectInstance)
-                    ? SerializeJSONArray(objectInstance)
-                    : SerializeJSONObject(objectInstance);
+                if (SerializesAsArray(objectInstance))
```
Comment (lahma) [inline Jint/Native/Json/JsonSerializer.cs:204]: nit: should have braces

**Review by lahma — APPROVED** [review]
This is quite cool and basically what I had in mind to support easier custom serialization, I like it.

**Comment by lahma** [comment]
Great addition, thank you!

**Comment by ajdust** [comment]
Thanks! And Thank you, @lahma for maintaining this great project 😄 👍

**Pull request merged**